### DISCUSSION TITLE
Why author did that?

### DISCUSSION THREAD

**Discussion by nezmogus** [farag2/Sophia-Script-for-Windows#483] [Ideas]
Hi,

why new versions of script does not have this command line inside script file? Each time this line should be copied from other files or typed manually...
**Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope Process -Force**

Why on any event script opens annoying and useless web pages, like telegram, discord, github? With each versions i should manually disable these code places inside script file.     

Thank you

**Comment by farag2** [comment]
Hi. I cannot add it into any script because this is how PowerShell works: we have to type this command manually before executing any script (depending on machine script execution policy of course).

> useless web pages, like telegram, discord, github

Because I have these community socials and I advertise them. You against it?) I won't be disabled by default.

**Reply by nezmogus** [reply]
This command was in earlier versions of script and was very easy to copy into power shell instead of typing manually each time. Why you removed it? Just leave as commented line inside script?

I understand your wish to advertise your social networks, and i wasn't against when these web pages popped out at the end of script execution, but now they are annoying. No internet - multiple web pages, new version  - multiple web pages, incompatible version  - multiple web pages, etc...

**Comment by farag2** [comment]
That was a mistake. :)

> No internet - multiple web pages, new version - multiple web pages, incompatible version - multiple web pages, etc...

I respect my users and if the script detects any issues it notifies user unlike any other script which supresses all errors or incompatibilities — they just do not care at all abiut users and their systems. I want users to know if anything goes wrong that why the script stops if there's a new version (whom I create a new versions if you do not what to use it?), any system break and others. Understood? 🐱

**Reply by nezmogus** [reply]
One thing is to notify user about error or new updates via error box or red line in power shell window (like it was in earlier script versions), and other thing is to open browser with multiple not necessary (and useless in most cases) web pages. If user found your script on the internet, he is quite smart to know, how to open browser manually and how to use google.

I personally hate apps, who automatically opens web browser without asking, like some apps do after install or uninstall...

**Reply by farag2** [reply]
> other thing is to open browser with multiple not necessary (and useless in most cases) web pages

They aren't useless. It iwll open it if user uses an old version.

> he is quite smart to know

I'd argue)))

>  who automatically opens web browser without asking

It opens browser only in case something is broken. Not because I want interrupt script work.

**Reply by nezmogus** [reply]
> > other thing is to open browser with multiple not necessary (and useless in most cases) web pages
> 
> They aren't useless. It iwll open it if user uses an old version.
Yes, they are. I counted at least 17 places where script opens web browser instead of showing error in power shell window. 

> 
> > he is quite smart to know
> 
> I'd argue)))
Ok, i agree there. But opened web pages definitely will not help for those users.
> 
> > who automatically opens web browser without asking
> 
> It opens browser only in case something is broken. Not because I want interrupt script work.

and the biggest pain is in words " **It opens browser**"...

**Reply by farag2** [reply]
Only in case if somethingis broken. Opening browser will be the least problem for user))

**Reply by nezmogus** [reply]
> Only in case if somethingis broken. Opening browser will be the least problem for user))

No. I always use script on freshly installed computer and script opens... EDGE(!!!) with all his stupid questions, also depending from OS version - opens edge in full screen and without close button. Then i should kill it in task manager to resolve script incompatibility and to get this same edge window again and again until all script incompatibility issues will be solved.

**Reply by farag2** [reply]
It doesn't open Edge just because. It opens wit purpose. What error do you get?

> script incompatibility

This is the answer. You were using unsupported Windows build. So?..

**Comment by AeliusSaionji** [comment]
In the Sophia.psm1, there is no singular function to comment out. Calls to invoke the browser are littered all over. Disabling this is a chore.
It would be nice if there were a single "do not open browser to or notify about social media" argument in the script.

But this behavior is incredibly misguided.

The script is intended to be edited by hand.
The social media urls are clearly and prominently placed at the top of the script file. Anyone editing the script will see them.

The script is intended to be run in the terminal.
Feel free to advertise the social media as terminal messages. No one will object to that.

You have said elsewhere that this script is intended for technical users.
Forcibly opening the browser and pinging notifications is not something an audience of technical users will appreciate. It seems like you expect the audience to be relatively nontechnical, exploring the terimnal for the first time.

If that is your audience, that's fine. But for that audience I strongly suggest you do NOT implicitly endorse disabling AdminApprovalMode and always running terminal as admin. The ability to do this should probably be removed from Sophia entirely, in my opinion. It's a terrible idea.

**Comment by nezmogus** [comment]
Because author refuses to take into account the opinions of users, for
personal use, I made cleaner on my web server. On sophia script zip file
upload it removes all annoying cr*p, updates config file with my preset
settings and returns new zip with *.bak files included (for comparison with
original files, just in case).

**Reply by farag2** [reply]
Do not mislead anyone who will be reading this thread: Sophia cannot do that.

> Feel free to advertise the social media as terminal messages. No one will object to that.

Just delete it and never use anymore, n00b.

**Reply by AeliusSaionji** [reply]
What's your website? It's not listed on your profile.
Do you have recommendations for similar projects?

**Reply by farag2** [reply]
Just do not use any of similar projects. Also, there's no similar projects, cus this one is the largest on GitHub.

**Comment by nezmogus** [comment]
As i wrote before, this is for my personal use. My web server isn't public.

**Comment by nezmogus** [comment]
Excuseme! Why we can't use any similar projects? :D
